sphinx.go

```diff
@@ -186,6 +192,11 @@ func generateSharedSecrets(paymentPath []*btcec.PublicKey,
 func NewOnionPacket(paymentPath *PaymentPath, sessionKey *btcec.PrivateKey,
 	assocData []byte) (*OnionPacket, error) {
 
+	// Check whether total payload size doesn't exceed the hard maximum.
+	if paymentPath.TotalPayloadSize() > routingInfoSize {
+		return nil, errMaxRoutingInfoSizeExceeded
```

Comment:
[Looking at the original issue again](https://github.com/lightningnetwork/lnd/issues/3712), I realized that this doesn't actually addresses the core panic: 
```
panic: runtime error: index out of range [20] with length 20

goroutine 8943415 [running]:
github.com/lightningnetwork/lnd/routing/route.(*Route).ToSphinxPath(0xc00648a140, 0x1da7740, 0xc010fbda00, 0x20)
        /home/bitcoin/go/src/github.com/lightningnetwork/lnd/routing/route/route.go:306 +0x53a
github.com/lightningnetwork/lnd/routing.generateSphinxPacket(0xc00648a140, 0xc0097def3c, 0x20, 0x20, 0xc01e500030, 0x0, 0xc00bede880, 0xf26542, 0xfc5f00, 0xc0000386f8, ...)
        /home/bitcoin/go/src/github.com/lightningnetwork/lnd/routing/router.go:1494 +0x4b
github.com/lightningnetwork/lnd/routing.(*paymentLifecycle).createNewPaymentAttempt(0xc0085c4f00, 0x40, 0x40, 0xc001034b80, 0xc01281b278, 0x52e342)
        /home/bitcoin/go/src/github.com/lightningnetwork/lnd/routing/payment_lifecycle.go:262 +0x475
github.com/lightningnetwork/lnd/routing.(*paymentLifecycle).resumePayment(0xc0085c4f00, 0x0, 0x0, 0x0, 0x0, 0x1, 0xb4e4ac, 0xc000010120)
        /home/bitcoin/go/src/github.com/lightningnetwork/lnd/routing/payment_lifecycle.go:52 +0xd0a
github.com/lightningnetwork/lnd/routing.(*ChannelRouter).sendPayment(0xc0004d0140, 0x0, 0xc0097def00, 0x14d6860, 0xc007ab2b40, 0x0, 0x0, 0x0, 0x0, 0x1151b00, ...)
```

The code panics here: https://github.com/lightningnetwork/lnd/blob/master/routing/route/route.go#L330 since the array is fixed size, and we're attempting to address an index outside of those bounds. This check will only be called after `ToSphinxPath()` is called, and actually try to make the final packet here: https://github.com/lightningnetwork/lnd/tree/master/https://github.com/lightningnetwork/lnd/blob/master/routing/route/route.go#L1514

So there's two checks needed (one here and one in lnd):
  * The total number of hops (not counting the payload size) isn't above `NumMaxHops`
  * The total number of serialized bytes for the packet encoding the mix packet itself, can fit into 1.3k bytes.

Reply:
Yes, this issue was fixed in this PR when I still had the variable path length. Now I indeed need to add another check for `SendToRoute`.

What I am also planning to do is add a running counter to path finding to make sure the payload never exceeds 1300 bytes. Especially with the custom records, this becomes more important to do. That automatically means that routes from path finding will never exceed 27 hops.